divi/src/masternode-payments.h

```diff
@@ -112,6 +139,8 @@ class CMasternodePaymentWinner
         READWRITE(nBlockHeight);
         READWRITE(payee);
         READWRITE(vchSig);
+        if (ser_action.ForRead())
+            ComputeSeedHash();
```

Comment:
Deserialization on read invoking the computation of the seed-hash could potentially result in a DDoS vector since the seed hash computation isn't cached. Better perhaps to just readwrite and soon after the start of the CMasternodePayments::ProcessMessageMasternodePayments method (for the case on mnwinners) you could have a public bool Verify() method that checks the hash perhaps as part of determining its validity.

Depending on the other changes this might not be releveant.

Reply:
This is a good point.  The only place where deserialisation happens is `ProcessMessageMasternodePayments`, but there are some more checks after the deserialisation in that function which we might as well do (and weed out some DoS attemps perhaps) before computing the hash.  I'll prepare an update that does the hash computation explicitly only right before it is needed (i.e. matching the behaviour before this PR).